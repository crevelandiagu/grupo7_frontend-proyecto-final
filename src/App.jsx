import { Typography, Button } from "@mui/material"
function App() {

  return (
    <>
      <Typography variant="h1">ABC Jobs</Typography>
      <Button variant="contained">Login</Button>
    </>
  )
}

export default App
