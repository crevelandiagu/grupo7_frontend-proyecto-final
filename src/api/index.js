export { default as candidateApi } from './candidateApi';
export { default as companyApi } from './companyApi';
export { default as companyEmployeesApi } from './companyEmployeesApi';
export { default as projectsApi } from './projectsApi';
export { default as searchToolApi } from './searchToolApi';
export { default as selectionProcess } from './selectionProcess';
export { default as performance } from './performanceApi';
