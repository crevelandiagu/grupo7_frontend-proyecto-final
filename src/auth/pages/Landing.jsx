import { Link as RouterLink } from 'react-router-dom';
import { styled } from '@mui/material/styles';
import Paper from '@mui/material/Paper';
import Button from '@mui/material/Button';
import CssBaseline from '@mui/material/CssBaseline';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box'; 
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import { createTheme, ThemeProvider } from '@mui/material/styles';


import candidateImg from '../../assets/candidate.svg';
import companiesImg from '../../assets/companies.svg';
import { getEnvVariablesCandidate, getEnvVariablesCompany, getEnvVariablesProject } from '../../helpers/getEnvVaribles';


const Item = styled(Paper)(({ theme }) => ({
  backgroundColor: theme.palette.mode === 'dark' ? '#1A2027' : '#fff',
  ...theme.typography.body2,
  padding: theme.spacing(1),
  textAlign: 'center',
  color: theme.palette.text.secondary,
}));

const defaultTheme = createTheme();

export const Landing = () => {

  return (
    <ThemeProvider theme={defaultTheme}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Box
          sx={{
            marginTop: 8,
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
          }}
        >
          <Typography component="h1" variant="h4">
            Select your profile 
            { console.log( import.meta.env.VITE_HOLA)}
          </Typography>
          <Box sx={{
            marginTop: 8,
            display: 'flex',
            flexDirection: 'row',
            alignItems: 'center',
          }}>
            <Grid container spacing={2}>
              <Grid item xs={6} >
                <Item>
                  <img src={candidateImg} width="200" alt=""/>
                </Item>
                <Button onClick={() => {localStorage.setItem('typeClient', 'company')}} component={ RouterLink } variant="outlined" to="/auth/signin">
                  You are a company
                </Button>
              </Grid>
              <Grid item xs={6} >
              <Item>
                <img src={companiesImg} width="200" alt="" />
              </Item>
              <Button onClick={() => {localStorage.setItem('typeClient', 'candidate')}} component={ RouterLink } variant="outlined" to="/auth/signin">
                  You are a candidate
                </Button>
              </Grid>
            </Grid>
          </Box>
        </Box>
      </Container>
    </ThemeProvider>
  );
}