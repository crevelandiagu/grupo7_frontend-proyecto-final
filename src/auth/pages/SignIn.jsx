import { useMemo, useState } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import Button from '@mui/material/Button';
import CssBaseline from '@mui/material/CssBaseline';
import TextField from '@mui/material/TextField';
import Link from '@mui/material/Link';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import { createTheme, ThemeProvider } from '@mui/material/styles';

import { useAuthStore, useForm } from '../../hooks';
import { Alert } from '@mui/material';

// import { checkingAuthentication } from '../../store/auth/thunks';

function Copyright(props) {
  return (
    <Typography variant="body2" color="text.primary" align="center" {...props}>
      {'Copyright Â© '}
      <Link component={ RouterLink } color="inherit" to ="landing">
        ABC Jobs
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const formData = {
  email:"",
  password: ""
}

const formValidations =  {
email: [ (value) => value.includes('@'), 'Enter a valid email' ],
password: [(value) => value.length >= 8, 'password must be at least 8 characters long']
}

const defaultTheme = createTheme();

export const SignIn = () => {

  const { startSignIn, errorMessage, status } = useAuthStore();
  const [formSubmitted, setFormSubmitted] = useState(false);

  const isCheckingAuthentication = useMemo( () => status === 'checking', [status]);
  // console.log('isCheckingAuthentication', isCheckingAuthentication, status === 'checking' )
  
  const {
    formState, email, password, onInputChange, isFormValid, 
    emailValid, passwordValid, } = useForm( formData, formValidations );
  // const dispatch =  useDispatch(); cuando se usan thunks

  const handleSubmit = (event) => {
    event.preventDefault();
    setFormSubmitted(true);
    // dispatch(checkingAuthentication()); cunado se usan thunks

    if ( !isFormValid ) return;
    startSignIn(formState);//({email, password});
    
    /*Navigate('/', {
      replace: true
    })
    */

  };

  return (
    <ThemeProvider theme={defaultTheme}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Box
          sx={{
            marginTop: 8,
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
          }}
        >
          <Typography component="h1" variant="h4">
            Welcome Back!
          </Typography>
          <Box component="form" onSubmit={handleSubmit} sx={{ mt: 3 }}>
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <TextField
                  label="Email Address"
                  type="email"
                  placeholder='yourmail@domain.com'
                  fullWidth
                  name="email"
                  value= {email}
                  onChange={onInputChange}
                  error = {!!emailValid && formSubmitted}
                  helperText = {emailValid}
                  // autoComplete="email"
                  // id="email"
                  // required
                />
              </Grid>
              <Grid item xs justifySelf="flex-end">
                <Link href="#" variant="body2">
                  Forgot password?
                </Link>
              </Grid>
              <Grid item xs={12}>
                <TextField
                  label="Password"
                  type="password"
                  placeholder="your password"
                  fullWidth
                  name="password"
                  value= {password}
                  onChange={onInputChange}
                  error = {!!passwordValid && formSubmitted}
                  helperText = {passwordValid}
                  // id="password"
                  // required

                />
              </Grid>
              
            </Grid>
            <Grid item sx={{ mt: 2 }}
              xs={12}
              display={ errorMessage ? '' : 'none' }
            >
                <Alert severity="error">{errorMessage}</Alert>
            </Grid> 
            <Button
              disabled = { isCheckingAuthentication }
              type="submit"
              fullWidth
              variant="contained"
              sx={{ mt: 3, mb: 2 }}
            >
              Sign In
            </Button>
            <Grid container>
              <Grid item>
                {"Don't have an account?"}
                <Link component={ RouterLink } variant="body2" to="/auth/signup">
                  {" Sign Up"}
                </Link>
                {/* <Link href="/auth/signup" >
                  con error
                </Link> */}
              </Grid>
            </Grid>
          </Box>
        </Box>
        <Copyright sx={{ mt: 5 }} />
      </Container>
    </ThemeProvider>
  );
}
