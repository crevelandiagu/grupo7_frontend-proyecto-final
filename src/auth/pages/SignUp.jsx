import { useMemo, useState } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import Button from '@mui/material/Button';
import CssBaseline from '@mui/material/CssBaseline';
import TextField from '@mui/material/TextField';
import Link from '@mui/material/Link';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import { Alert } from '@mui/material';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import { createTheme, ThemeProvider } from '@mui/material/styles';

import { useAuthStore, useForm } from '../../hooks';


function Copyright(props) {
  return (
    <Typography variant="body2" color="text.primary" align="center" {...props}>
      {'Copyright Â© '}
      <Link component={ RouterLink } color="inherit" to ="landing">
        ABC Jobs
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const formData = {
  email:"",
  password: ""
}

const formValidations =  {
email: [ (value) => value.includes('@'), 'Enter a valid email' ],
password: [(value) => value.length >= 6, 'password must be at least 8 characters long']
}

const defaultTheme = createTheme();

export const SignUp = () => {

  const { startSignUp, errorMessage, status } = useAuthStore();
  const [formSubmitted, setFormSubmitted] = useState(false);
  
  const isCheckingAuthentication = useMemo( () => status === 'checking', [status]);
  //console.log('isCheckingAuthentication', isCheckingAuthentication, status === 'checking' )
  
  const {
    formState, email, password, onInputChange, isFormValid, 
    emailValid, passwordValid, } = useForm( formData, formValidations );

  const handleSubmit = (event) => {
    event.preventDefault();
    setFormSubmitted(true);
    //console.log(formState, typeof isFormValid, isFormValid, !!emailValid && formSubmitted)
    
    if ( !isFormValid ) return;
    startSignUp(formState);//({email, password});
  };

  return (
    <ThemeProvider theme={defaultTheme}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Box
          sx={{
            marginTop: 8,
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
          }}
        >
          <Typography component="h1" variant="h4">
            Create an account
          </Typography>
          <Box component="form" onSubmit={handleSubmit} sx={{ mt: 3 }}>
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <TextField
                  label="Email Address"
                  type="email"
                  placeholder='your-email@domain.com'
                  fullWidth
                  name="email"
                  value= {email}
                  onChange={onInputChange}
                  error = {!!emailValid && formSubmitted}
                  helperText = {emailValid}
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  label="Password"
                  type="password"
                  placeholder="your password"
                  fullWidth
                  name="password"
                  value= {password}
                  onChange={onInputChange}
                  error = {!!passwordValid && formSubmitted}
                  helperText = {passwordValid}
                />
              </Grid>
              
            </Grid>
            <Grid item sx={{ mt: 2 }}
              xs={12}
              display={ errorMessage ? '' : 'none' }
            >
                <Alert severity="error">{errorMessage}</Alert>
            </Grid> 
            <Button
              disabled = { isCheckingAuthentication }
              type="submit"
              fullWidth
              variant="contained"
              sx={{ mt: 3, mb: 2 }}
            >
              Sign Up
            </Button>
            <Grid container justifyContent="flex-start">
              <Grid item>
                By signing up, you agree to our Terms & Conditions.
              </Grid>
              <Grid item>
                {`Already have an account? `}
                <Link component={ RouterLink } variant="body2" to="/auth/signin">
                  {" Sign In"}
                </Link>
              </Grid>
            </Grid>
          </Box>
        </Box>
        <Copyright sx={{ mt: 5 }} />
      </Container>
    </ThemeProvider>
  );
}
