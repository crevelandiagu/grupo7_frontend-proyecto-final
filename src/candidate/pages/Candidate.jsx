import { useSelector } from 'react-redux';
import { CandidateLayout } from '../layout/CandidateLayout';
import { CandidateDashboard, Interview, PerformanceList, ProfileBasic, ProfileCertificates, ProfileEducation, ProfileExperience, SignContract } from '../views';
import { AssesmentList } from '../views/AssesmentList';

export const Candidate = () => {

  const {view} = useSelector( state => state.candidate );

  const children = view === 'dashboard' &&<CandidateDashboard />
  || view === 'basic' && <ProfileBasic />
  || view === 'certificates' && <ProfileCertificates />
  || view === 'education' && <ProfileEducation />
  || view === 'experience' && <ProfileExperience />
  || view === 'interview' && <Interview />
  || view === 'signcontract' && <SignContract />
  || view === 'assesmentList' && <AssesmentList />
  || view === 'performanceList' && <PerformanceList />

  return (
    <CandidateLayout>
      { 
        children
      }
    </CandidateLayout>
  )
}