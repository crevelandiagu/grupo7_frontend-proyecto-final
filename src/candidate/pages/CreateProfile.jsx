export const CreateProfile = () => {

  return (
    <>
      <h1>CreateProfile</h1>
    </>
  )

}