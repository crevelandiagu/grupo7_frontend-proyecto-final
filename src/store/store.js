import { configureStore } from '@reduxjs/toolkit'
import { authSlice } from './auth'
import { candidateSlice } from './candidate/candidateSlice'

export const store =  configureStore({
  reducer: {
    auth: authSlice.reducer,
    candidate: candidateSlice.reducer,
  },
})