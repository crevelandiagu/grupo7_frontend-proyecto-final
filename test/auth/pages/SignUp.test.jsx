import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../src/store';
import { SignUp } from '../../../src/auth/pages/SignUp';
import { MemoryRouter } from 'react-router';

describe('Test to componenet <Landing />', () => {
  test('Should be match with snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
        <SignUp />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  })
});