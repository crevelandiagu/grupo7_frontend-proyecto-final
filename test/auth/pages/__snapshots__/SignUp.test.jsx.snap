// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to componenet <Landing /> Should be match with snapshot 1`] = `
<div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-binzgt"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Create an account
      </h1>
      <form
        class="MuiBox-root css-178yklu"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r0:"
                id=":r0:-label"
              >
                Email Address
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  name="email"
                  placeholder="your-email@domain.com"
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Email Address
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r1:"
                id=":r1:-label"
              >
                Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r1:"
                  name="password"
                  placeholder="your password"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1vddya1-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-1hdyuqk-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="ErrorOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Sign Up
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiGrid-root MuiGrid-container css-1bzpx2o-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            By signing up, you agree to our Terms & Conditions.
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            Already have an account? 
            <a
              class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-101ca9i-MuiTypography-root-MuiLink-root"
              href="/auth/signin"
            >
               Sign In
            </a>
          </div>
        </div>
      </form>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-4cdb0b-MuiTypography-root"
    >
      Copyright Â© 
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-xdxgn1-MuiTypography-root-MuiLink-root"
        href="/landing"
      >
        ABC Jobs
      </a>
       
      2023
      .
    </p>
  </main>
</div>
`;
